### ISSUE TITLE
Navbar "inverted" Text Color

### ISSUE THREAD

**Issue by endeffects** [chanan/BlazorStrap#97]
When i set the Navbar Color to Dark, should the Text Color not be adjusted automatically?
Or do i have to override this with css?

`<Navbar IsExpand="true" IsFixedTop="true" Color="Color.Dark" class="navbar-expand-md">`

**Comment by chucker** [comment]
Judging from the code, it seems you need to additionally set `IsDark="true"`.

IMHO, we should change the setter such that if Color == Color.Dark, IsDark automatically gets set to true. (The reverse doesn't hold, though: [you should be able to combine Dark with e.g. Primary](https://getbootstrap.com/docs/4.3/components/navbar/#color-schemes).)

**Comment by endeffects** [comment]
Thanks alot, is there any overview for all this switches?

**Comment by endeffects** [comment]
By the way, this Flag does not effect the DropDownToggles. The Text Color is still dark.

```
<Navbar IsExpand="true" IsFixedTop="true" IsDark="true" Color="Color.Dark" class="navbar-expand-md">
    <NavbarBrand href="#">Brand</NavbarBrand>
    <NavbarToggler onclick="@(async () => await OnClick())" />
    <Collapse isOpen="@IsOpen" IsNavbar="true">
        <BlazorNav IsList="true" IsNavbar="true">
            <NavItem>
                <BlazorNavLink IsActive="true" href="#" Match="NavLinkMatch.All">Home</BlazorNavLink>
            </NavItem>
            <NavItem IsDropdown="true">
                <DropdownToggle IsDark="true">CategoryOne</DropdownToggle>
                <DropdownMenu>
                    <DropdownItem href="CategoryOne/Create">Create</DropdownItem>
                    <DropdownItem href="CategoryOne/Delete">Delete</DropdownItem>
                    <DropdownItem href="CategoryOne/Rename">Rename</DropdownItem>
                    <DropdownItem href="CategoryOne/Show">Show</DropdownItem>
                </DropdownMenu>
            </NavItem>
            <NavItem IsDropdown="true">
                <DropdownToggle IsDark="true">CategoryTwo</DropdownToggle>
                <DropdownMenu>
                    <DropdownItem href="CategoryTwo/Create">Create</DropdownItem>
                    <DropdownItem href="CategoryTwo/Delete">Delete</DropdownItem>
                    <DropdownItem href="CategoryTwo/Rename">Rename</DropdownItem>
                    <DropdownItem href="CategoryTwo/Show">Show</DropdownItem>
                </DropdownMenu>
            </NavItem>
        </BlazorNav>
    </Collapse>
</Navbar>
```

**Comment by chucker** [comment]
I don't believe so. I looked at the source code for [Navbar](https://github.com/chanan/BlazorStrap/blob/master/src/BlazorStrap/Navbar.razor).

**Comment by endeffects** [comment]
Using the suggested styling on the navbar directly also does not fix the dropdown boxes

`<Navbar IsExpand="true" IsFixedTop="true" class="navbar-expand-md navbar-dark bg-dark">`

![image](https://user-images.githubusercontent.com/13432569/60890598-e3422f80-a25b-11e9-8b60-ee18aa801f54.png)

**Comment by chanan** [comment]
I will re-check the source project (react-strap) and see if I missed something

**Comment by chanan** [comment]
I couldn't find a dropdown in nav in react-strap or in the actual boot strap site. I did find one here though: https://react-bootstrap.github.io/components/navbar/#navbars-mobile-friendly

Looking at the generated CSS, its different from the way we currently handle dropdown in navs. So to fix this in that way I would need to do some writing of the code.

One thing you can do if you want to use this, is pass you one class or even your own style for now using the Class or style attributes on DropdownToggle

**Comment by endeffects** [comment]
I think the problem is related to the dropdown toggle button only.

**Comment by chanan** [comment]
It is.
but there is two ways to fix it, one is just apply a color to the dropdown toggle which i am not crazy about, and the other is to use the correct bootstrap css and structure. The only example i found so far has the html elements in a different way then I have them:

Closed:
```
<div class="dropdown nav-item"><a aria-haspopup="true" aria-expanded="false" href="#" class="dropdown-toggle nav-link" role="button">Dropdown</a><div class="dropdown-menu" aria-labelledby=""><a href="#action/3.1" class="dropdown-item">Action</a><a href="#action/3.2" class="dropdown-item">Another action</a><a href="#action/3.3" class="dropdown-item">Something</a><div class="dropdown-divider" role="separator"></div><a href="#action/3.4" class="dropdown-item">Separated link</a></div></div>
```

Open:
```
<div class="show dropdown nav-item"><a aria-haspopup="true" aria-expanded="true" href="#" class="dropdown-toggle nav-link" role="button">Dropdown</a><div class="dropdown-menu show" aria-labelledby=""><a href="#action/3.1" class="dropdown-item">Action</a><a href="#action/3.2" class="dropdown-item">Another action</a><a href="#action/3.3" class="dropdown-item">Something</a><div class="dropdown-divider" role="separator"></div><a href="#action/3.4" class="dropdown-item">Separated link</a></div></div>
```

So that is the fix that I am going to apply, but it will take a me a bit to do it.

(If you want to submit a PR for the above, please do)

**Comment by endeffects** [comment]
I'm pretty new on all this bootstrap stuff, so i'm not sure if i can provide usefull fixes here. :(

**Comment by chanan** [comment]
No worries, I will take attempt to figure out why my version is different from react-bootstrap later this week.

**Comment by chanan** [comment]
will be released in 1.0.0-preview6-21

The api itself is not great... But I dont want to tackle cleaning up and adding better documentation as part of this issue. For now, I will paste the code I used here:

```
<Navbar IsExpand="true" IsFixedTop="true" IsDark="true" Color="Color.Dark" class="navbar-expand-md">
    <NavbarBrand href="#">Brand</NavbarBrand>
    <NavbarToggler />
    <Collapse isOpen="true" IsNavbar="true">
        <BlazorNav IsNavbar="true">
            <NavItem>
                <BlazorNavLink IsActive="true" href="#" Match="NavLinkMatch.All">Home</BlazorNavLink>
            </NavItem>
            <NavItem IsDropdown="true">
                <DropdownToggle IsDark="true" IsLink="true">CategoryOne</DropdownToggle>
                <DropdownMenu>
                    <DropdownItem href="CategoryOne/Create">Create</DropdownItem>
                    <DropdownItem href="CategoryOne/Delete">Delete</DropdownItem>
                    <DropdownItem href="CategoryOne/Rename">Rename</DropdownItem>
                    <DropdownItem href="CategoryOne/Show">Show</DropdownItem>
                </DropdownMenu>
        </BlazorNav>
    </Collapse>
</Navbar>
```

The "important part" is `IsLink="true"` on `DropdownToggle`. Theoretically it should be the correct color as a button as well, the problem is I haven't found an example in the wild with a dark nav and a button dropdown as mentioned in my previous comments. So till we do, we will need to use link dropdowns (which only work in 1.0.0-preview6-21)

**Comment by endeffects** [comment]
Thank you for your help :)
